caravel/assets/javascripts/dashboard/Dashboard.jsx

```diff
@@ -39,16 +40,30 @@ var Dashboard = function (dashboardData) {
       });
       this.slices = sliceObjects;
       this.refreshTimer = null;
+      this.loadPreSelectFilters();
       this.startPeriodicRender(0);
       this.bindResizeToWindowResize();
     },
-    setFilter: function (slice_id, col, vals) {
-      this.addFilter(slice_id, col, vals, false);
+    loadPreSelectFilters: function () {
+      try {
+        var filters = JSON.parse(px.getParam("preselect_filters") || "{}");
+        for (var slice_id in filters) {
+          for (var col in filters[slice_id]) {
+            this.setFilter(slice_id, col, filters[slice_id][col], false, false);
+          }
+        }
+      } catch (e) {
+        console.error(e);
+      }
+    },
+    setFilter: function (slice_id, col, vals, refresh) {
+      this.addFilter(slice_id, col, vals, false, refresh);
     },
-    addFilter: function (slice_id, col, vals, merge) {
+    addFilter: function (slice_id, col, vals, merge, refresh) {
```

Comment:
I just realized that ES6 supports function default params much like in python. We should use it starting now.
https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Default_parameters